services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/ScanEnhancedRequest.java

```diff
@@ -35,14 +38,14 @@
     private final Integer limit;
     private final Boolean consistentRead;
     private final Expression filterExpression;
-    private final String projectionExpression;
+    private final List<String> attributesToProject;
 
     private ScanEnhancedRequest(Builder builder) {
         this.exclusiveStartKey = builder.exclusiveStartKey;
         this.limit = builder.limit;
         this.consistentRead = builder.consistentRead;
         this.filterExpression = builder.filterExpression;
-        this.projectionExpression = builder.projectionExpression;
+        this.attributesToProject = builder.attributesToProject;
```

Comment:
We should ensure this list cannot be modified, especially as we're vending it in a getter. I suggest ```this.attributesToProject = Collections.unmodifiableList(builder.attributesToProject);```